lib/kubernetes-deploy/runner.rb

```diff
@@ -301,24 +303,7 @@ def confirm_namespace_exists
     end
 
     def run_kubectl(*args, namespaced: true, with_context: true)
-      args = args.unshift("kubectl")
-      if namespaced
-        raise FatalDeploymentError, "Namespace missing for namespaced command" if @namespace.blank?
-        args.push("--namespace=#{@namespace}")
-      end
-
-      if with_context
-        raise FatalDeploymentError, "Explicit context is required to run this command" if @context.blank?
-        args.push("--context=#{@context}")
-      end
-      KubernetesDeploy.logger.debug Shellwords.join(args)
-      out, err, st = Open3.capture3(*args)
-      KubernetesDeploy.logger.debug(out.shellescape)
-      unless st.success?
-        KubernetesDeploy.logger.warn("The following command failed: #{Shellwords.join(args)}")
-        KubernetesDeploy.logger.warn(err)
-      end
-      [out.chomp, err.chomp, st]
+      super(*args, namespace: @namespace, context: @context, namespaced: namespaced, with_context: with_context)
```

Comment:
TBH I see it a bit confusing to use the super method that comes from the included module... I think it would be more clear to explicitly call the helper: `Kubectl.build_command(*args, namespace: foo, context: blah)`